src/app/grid/grid-validator-service-extended/grid-validator-service-extended.component.ts

```diff
@@ -0,0 +1,100 @@
+import { Component, Directive, Input, ViewChild } from '@angular/core';
+import { AbstractControl, FormGroup, NG_VALIDATORS, ValidationErrors, ValidatorFn } from '@angular/forms';
+import { IgxColumnValidator, IgxGridComponent } from 'igniteui-angular';
+import { employeesData } from '../../data/employeesData';
+
+export function forbiddenNameValidator(nameRe: RegExp): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+        const forbidden = nameRe.test(control.value);
+        return forbidden ? { forbiddenName: { value: control.value } } : null;
+    };
+}
+
+@Directive({
+    selector: '[forbiddenName]',
+    providers: [{ provide: NG_VALIDATORS, useExisting: ForbiddenValidatorDirective, multi: true }]
+})
+export class ForbiddenValidatorDirective extends IgxColumnValidator {
+    @Input('forbiddenName')
+    public forbiddenNameString = '';
+
+    public validate(control: AbstractControl): ValidationErrors | null {
+        return this.forbiddenNameString ? forbiddenNameValidator(new RegExp(this.forbiddenNameString, 'i'))(control)
+            : null;
+    }
+}
+
+@Component({
+    selector: 'app-grid-validator-service-extended',
+    styleUrls: ['./grid-validator-service-extended.component.scss'],
+    templateUrl: './grid-validator-service-extended.component.html'
+})
+export class GridValidatorServiceExtendedComponent {
+
+    @ViewChild('grid1', { read: IgxGridComponent })
+    public grid: IgxGridComponent;
+
+    public data: any[];
+    public employeesData: any[];
+    public transactionData = JSON.parse(JSON.stringify(employeesData));
+
+    constructor() {
+        this.data = employeesData;
+    }
+
+    public formCreateHandler(formGroup: FormGroup) {
+        const faxRecord = formGroup.get('fax');
+        const createdOnRecord = formGroup.get('created_on');
+        const lastActiveRecord = formGroup.get('last_activity');
+        faxRecord.addValidators(this._testFax());
+        createdOnRecord.addValidators(this._testDateRecord());
+        lastActiveRecord.addValidators(this._testDateRecord(createdOnRecord.value));
```

Comment:
Validators get cached, so this value that is explicitly passed to the function will not get updated if you enter to edit the same row a second time. Would be better to just get the values you want to compare inside the validator function itself via the formGroup, for example:
```
return (control: AbstractControl): ValidationErrors | null => {
const formGroup = control.parent;
const value1 = formGroup.get('createdOnRecord');
const value2 = formGroup.get('lastActiveRecord');
...
```
As these will always be up-to-date.